x-pack/osquerybeat/ext/osquery-extension/pkg/amcache/tables/application.go

```diff
@@ -0,0 +1,109 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build windows
+
+package tables
+
+import (
+	"context"
+	"github.com/elastic/beats/v7/x-pack/osquerybeat/ext/osquery-extension/pkg/encoding"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+// ApplicationEntry represents a single entry in the amcache application table.
+type ApplicationEntry struct {
+	LastWriteTime      int64  `osquery:"last_write_time"`
+	ProgramId          string `osquery:"program_id"`
+	ProgramInstanceId  string `osquery:"program_instance_id"`
+	Name               string `osquery:"name"`
+	Version            string `osquery:"version"`
+	Publisher          string `osquery:"publisher"`
+	Language           int64  `osquery:"language"`
```

Comment:
this type seems to mismatch the column type which is a textcolumn below, you probably can integrate what I added https://github.com/elastic/beats/pull/47117/files#diff-92ff53a6942ba8ea1d63bdd443e1b6c66726aee14b494f28182457b1cf583bbb, was not planning to do it in a separate PR but if it is best I can just split it